sklearn/metrics/tests/test_pairwise.py

```diff
@@ -370,6 +372,49 @@ def test_pairwise_distances_argmin_min():
     np.testing.assert_almost_equal(dist_orig_val, dist_chunked_val, decimal=7)
 
 
+def check_invalid_block_size_generator(generator):
+    for i in generator:
+        return i
+
+
+def test_pairwise_distances_blockwise_invalid_block_size():
+    rng = np.random.RandomState(0)
+    X = rng.random_sample((400, 4))
+    y = rng.random_sample((200, 4))
+    gen = pairwise_distances_blockwise(X, y, block_size=0, metric='euclidean')
+    assert_raise_message(ValueError, 'block_size should be at least n_samples '
+                         '* 8 bytes = 1 MiB, got 0',
+                         check_invalid_block_size_generator, gen)
+
+
+def test_pairwise_distances_blockwise():
+    # Test the pairwise_distance helper function.
+    rng = np.random.RandomState(0)
+    # Euclidean distance should be equivalent to calling the function.
+    X = rng.random_sample((400, 4))
+    gen = pairwise_distances_blockwise(X, block_size=1, metric="euclidean")
```

Comment:
we should be testing at least one other metric. I think you would be best with a helper function like `check_pairwise_distances_blockwise`, then call it for different sets of parameters inclduing metric, `Y` None or otherwise, and `block_size`